Add changeId to commit metadata. (#1049)
projectnessie/nessie: site/docs/develop/spec.md

There is no API to look up an object by `id` and the intention of an `id` is not to serve in that capacity. An example usage
of the `id` field might be storing auxiliary data on an object in a local cache and using `id` to look up that auxiliary data.

#### PermanentId

All contents object must have a `permanentId` field. This field is set at commit time and is unique for the commit. All objects in
the commit will have the same `permanentId`. This field would not be updated on merges or transplants therefore the id is unique for
that commit permanently. By convention it is a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) though
this is not enforced by the Specification. There are several expectations on this field:

1. `permanentId`  will always be updated at commit time.

> 1. The `permanentId` will not be updated during any transplant or merge operations.

Reviewer: ```suggestion
1. `permanentId` will not be updated during any transplant or merge operations.
```

1. All objects in the same original commit will have the same `permanentId`

> 1. This has **nothing** to do with the commit hash.

Reviewer: ```suggestion
1. `permanentId` is *not* related or derived from the commit-hash.
```

There is no API to look up an object by `permanentId` and the intention of an `permanentId` is not to serve in that capacity.

> An example usage of the `permanentId` field might be storing auxiliary data on an object in a local cache and using

Reviewer: ```suggestion
An example usage of the `permanentId` field is storing auxiliary data on an object in a local cache and using
```

`permanentId` to look up that auxiliary data.

!!! note
    A note about caching. The `Contents` objects are likely to be cached locally by services using Nessie. There is also likely
    to be auxiliary data (eg schema, partitions etc) stored by services which refer to a specific `Contents` at a specific

>     commit or time. The tuple `(id, permanentId)` uniquely reference an object in the Nessie history and is a suitable key

Reviewer: Can you elaborate where "id" comes from?

Author: fixed
